src/EditorFeatures/Core/SymbolSearch/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -110,19 +104,19 @@ public Updater(SymbolSearchUpdateEngine service, string source, string localSett
                 }
 
                 // Keep on looping until we're told to shut down.
-                while (!_service._updateCancellationToken.IsCancellationRequested)
+                while (!cancellationToken.IsCancellationRequested)
                 {
-                    await _service.LogInfoAsync("Starting update").ConfigureAwait(false);
+                    await _service.LogInfoAsync("Starting update", cancellationToken).ConfigureAwait(false);
                     try
                     {
-                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync().ConfigureAwait(false);
+                        var delayUntilNextUpdate = await UpdateDatabaseInBackgroundWorkerAsync(cancellationToken).ConfigureAwait(false);
 
-                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update").ConfigureAwait(false);
-                        await Task.Delay(delayUntilNextUpdate, _service._updateCancellationToken).ConfigureAwait(false);
+                        await _service.LogInfoAsync($"Waiting {delayUntilNextUpdate} until next update", cancellationToken).ConfigureAwait(false);
+                        await Task.Delay(delayUntilNextUpdate, cancellationToken).ConfigureAwait(false);
                     }
                     catch (OperationCanceledException)
                     {
-                        await _service.LogInfoAsync("Update canceled. Ending update loop").ConfigureAwait(false);
+                        await _service.LogInfoAsync("Update canceled. Ending update loop", cancellationToken).ConfigureAwait(false);
```

Comment:
in existing design. cancellation is simple.

VS side, session owns cancellation. and raising cancellation close connection to service.
service side. connection owns cancellation. when connection is closed, cancellation will be raised.

in existing design. service side can never raise cancellation. if service side closed the connection, then it is a bug - VS will show info bar to users saying one should restart VS.